Modules/Core/Common/include/itkImageBase.h

```diff
@@ -768,6 +768,19 @@ class ITK_TEMPLATE_EXPORT ImageBase : public DataObject
   bool
   VerifyRequestedRegion() override;
 
+  /** Check whether this image and the other image occupy the same physical space.
+   * Compares the origin, spacing, and direction for equality within provided tolerances. */
+  bool
+  SamePhysicalSpaceAs(const ImageBase * otherImage, double coordinateTolerance, double directionTolerance) const;
+
+  /** Check whether this image and the other image have the same grid in physical space.
+   * Compares largest possible regions for equality, and the origin, spacing,
+   * and direction cosines for equality within provided tolerances. */
+  bool
+  SameImageGridAs(const ImageBase * otherImage,
+                  double            coordinateTolerance = 1e-6,
+                  double            directionTolerance = 1e-6) const;
```

Comment:
Is this default of `1e-6` arbitrary? If so, what about using `DBL_EPSILON` instead, as a default? Personally I would prefer zero-tolerance (`= 0.0`) by default.